### PR TITLE
Catch only non-fatal errors loading config

### PR DESCRIPTION
**Pull request by bcarter97**
## Description

- Throw fatal error as suggested here: https://github.com/alexandru/scala-best-practices/blob/master/sections/2-language-rules.md#28-must-not-catch-throwable-when-catching-exceptions
- Fix config test name

### CODE REVIEW

File: src/main/scala/com/sky/kafka/topicloader/config/config.scala [hunk 1]
```diff
@@ -16,7 +17,8 @@ package object config {
   implicit class EitherOps[A](e: Either[Throwable, A]) {
```
Comment (lacarvalho91) [inline src/main/scala/com/sky/kafka/topicloader/config/config.scala:17]: this is only called in `Config.load` for `PosInt`s. So I would either change this syntax extension so it takes an `Either[IllegalArgumentException, A]` or I'd change the `PosInt` apply to return `Either[ConfigException.BadValue, PosInt]`

Then you don't have to worry about catching `Throwable` anywhere.

If you go with the option of changing the syntax extension method, I think it should also be made package private (I dunno if it was private before though, or if it even existed before)

File: src/main/scala/com/sky/kafka/topicloader/config/config.scala [hunk 2]
```diff
@@ -16,7 +17,8 @@ package object config {
...
     def validate(path: String): ValidationResult[A] = e.leftMap {
       case ce: ConfigException => ce
-      case e: Throwable        => new ConfigException.BadValue(path, e.getMessage)
+      case e if NonFatal(e)    => new ConfigException.BadValue(path, e.getMessage)
```
Comment (TobyPinfold) [inline src/main/scala/com/sky/kafka/topicloader/config/config.scala:20]: I think you can shorten this to just be
```suggestion
      case NonFatal(e)  => new ConfigException.BadValue(path, e.getMessage)
```

File: src/main/scala/com/sky/kafka/topicloader/config/config.scala (hunk 1 above)
Comment (bcarter97) [reply src/main/scala/com/sky/kafka/topicloader/config/config.scala]: Went with option 1, made it private. Still have to catch a nonFatal in the TryOps as the Try will always throw a ConfigException but it's unboxed so have to re-wrap it

**Review by lacarvalho91 — APPROVED** [review]
